### PR TITLE
user blocking of domains

### PR DESCRIPTION
**Pull request by daliacoss**
this is a patch for issue #423. it's still in the early stages as it only implements the AccountDomainBlock model, and filtering out toots by blocked domains from the user's public timeline. the finished patch will update the API and provide an interface for the user to create domain blocks.

this builds off of @hach-que 's unfinished patch #302.

### CODE REVIEW

File: app/controllers/api/v1/accounts_controller.rb [hunk 1]
```diff
@@ -85,7 +85,7 @@ def unblock
   def relationships
     ids = params[:id].is_a?(Enumerable) ? params[:id].map(&:to_i) : [params[:id].to_i]
 
-    @accounts    = Account.where(id: ids).select('id')
+    @accounts    = Account.where(id: ids).select('id, domain')
```
Comment (Gargron) [inline app/controllers/api/v1/accounts_controller.rb]: Dunno what you're doing here tho!

Comment (deckycoss) [inline app/controllers/api/v1/accounts_controller.rb]: i think i was testing something that i didn't quite understand from @hach-que 's closed patch, and forgot to revert it

File: app/controllers/api/v1/domains_controller.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    # helper object as there is currently no Domain model
+
+    attr_accessor :name, :current_account
+    
+    def initialize(name, current_account)
+        @name = name
+        @current_account = current_account
+    end
+    
+    def blocked
```
Comment (Gargron) [inline app/controllers/api/v1/domains_controller.rb]: If it returns a boolean, the convention is to end the method name with `?`

**Comment by deckycoss** [comment]
@Gargron see my last few messages on masto

**Comment by deckycoss** [comment]
@Gargron i think i finally managed to merge from upstream without breaking anything. this patch is done until i receive any feedback suggesting otherwise.

by the way, i never figured out why the tests originally passed "blocking?" an array, since whenever i do this the result is always false, even if the array matches the set of accounts that are being blocked. i was able to make the test failures go away by adding `return false unless other_account.is_a?(Account)` to my "blocking_account_or_domain?" method, thus preventing arrays from being used at all, but that felt like a weird solution so i haven't pushed it yet.

**Comment by tobascodagama** [comment]
Is there a reason this was closed?

**Comment by staticsafe** [comment]
This issue is being tackled in issue #2381. You can follow that one for updates.

**Pull request closed without merging**